domain-model-assistant/Assets/Components/Scripts/TextBox.cs

```diff
@@ -34,19 +58,72 @@ void Update()
         }
     }
 
-    public bool SetSection(GameObject section)
+    public bool isValid(){
+        bool res = false;
+        string text = this.GetComponent<InputField>().text;
+        //check that inputfield is of a particular format (Eg. int year)
+        string[] values = text.Split(' ');
+        Debug.Log(values.Length);
+        if (values.Length == 2 && !String.IsNullOrWhiteSpace(values[1]))
+        {
+            // if(!values[1].Equals(' ')
+            Debug.Log("second element is: " + values[1]);
+            this.SetTypeId(values[0]);
+            this.SetName(values[1]);
+            return res = true;
+        } 
+        else 
+        {
+            return res = false;
+        }
+    }
+
+    public void SetName(string aName)
+    {
+        this.name = aName;
+    }
+
+    public string GetName()
+    {
+        return this.name;
+    }
+
+    public void SetTypeId (string value)
+    {
+        Dictionary<string, string> tmpDict = _diagram.getAttrTypeIdsToTypes();
+        string tmpTypeId = null;
+        foreach(var item in tmpDict)
+        {
+            if (item.Value.Equals(value))
+            {
+                tmpTypeId = item.Key;
+                break;
+            }
+        }
+        if (!String.IsNullOrEmpty(tmpTypeId))
+        {
+            this.typeId = Int16.Parse(tmpTypeId);
+        } 
+    }
+
+    public int GetTypeId()
+    {
+        return this.typeId;
+    }
+        
+    public bool SetSection(GameObject sSection)
     {
-        if (section == null)
+        if (sSection == null)
         {
             return false;
         }
-        this.section = section;
+        this.section = sSection;
         return true;
     }
 
     public GameObject GetSection()
     {
-        return section;
+        return this.section;
     }
 
     public void OnBeginHold2()
```

Comment:
Why is there a "2" at the end of this method name? Rename this or at least add a comment to make this clear

Reply:
Fixed